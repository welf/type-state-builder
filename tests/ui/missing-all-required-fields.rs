use type_state_builder::TypeStateBuilder;

#[derive(TypeStateBuilder)]
struct Config {
    #[builder(required)]
    api_key: String,
    
    #[builder(required)]
    endpoint: String,
    
    timeout: Option<u32>,
}

fn main() {
    // This should fail: trying to build without setting any required fields
    let config = Config::builder().build(); // Error: missing required fields 'api_key' and 'endpoint'
}

