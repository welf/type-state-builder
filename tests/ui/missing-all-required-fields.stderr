error[E0599]: no method named `build` found for struct `ConfigBuilder_MissingApiKey_MissingEndpoint` in the current scope
  --> tests/ui/missing-all-required-fields.rs:18:10
   |
3  |   #[derive(TypeStateBuilder)]
   |            ---------------- method `build` not found for this struct
...
16 |       let config = Config::builder()
   |  __________________-
17 | |         .timeout(Some(5000))
18 | |         .build(); // Error: missing required fields 'api_key' and 'endpoint'
   | |         -^^^^^ method not found in `ConfigBuilder_MissingApiKey_MissingEndpoint`
   | |_________|
   |
